libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -1105,68 +1043,53 @@ void AC_PosControl::rate_to_accel_xy(float dt, float ekfNavVelGainScaler)
-        _accel_target_filter.reset(Vector2f(_accel_target.x, _accel_target.y));
-        _flags.reset_accel_to_lean_xy = false;
-    }
+    // pass the correction acceleration to the target acceleration output
+    _accel_target.x = _accel_target_filter.get().x;
+    _accel_target.y = _accel_target_filter.get().y;
 
-    // apply jerk limit of 17 m/s^3 - equates to a worst case of about 100 deg/sec/sec
-    float max_delta_accel = dt * _jerk_cmsss;
+    // Add feed forward into the target acceleration output
+    _accel_target.x += _accel_desired.x;
+    _accel_target.y += _accel_desired.y;
 
-    Vector2f accel_in(_accel_target.x, _accel_target.y);
-    Vector2f accel_change = accel_in-_accel_target_jerk_limited;
-    float accel_change_length = accel_change.length();
+    // the following section converts desired accelerations provided in lat/lon frame to roll/pitch angles
 
-    if(accel_change_length > max_delta_accel) {
-        accel_change *= max_delta_accel/accel_change_length;
-    }
-    _accel_target_jerk_limited += accel_change;
+    // limit acceleration using maximum lean angles
+    float angle_max = MIN(_attitude_control.get_althold_lean_angle_max(), get_lean_angle_max_cd());
+    float accel_max = MIN(GRAVITY_MSS * 100.0f * tanf(ToRad(angle_max * 0.01f)), POSCONTROL_ACCEL_XY_MAX);
+    _limit.accel_xy = limit_vector_length(_accel_target.x, _accel_target.y, accel_max);
 
-    // lowpass filter on NE accel
-    _accel_target_filter.set_cutoff_frequency(MIN(_accel_xy_filt_hz, 5.0f*ekfNavVelGainScaler));
-    Vector2f accel_target_filtered = _accel_target_filter.apply(_accel_target_jerk_limited, dt);
+    // update angle targets that will be passed to stabilize controller
+    accel_to_lean_angles(_accel_target.x, _accel_target.y, _roll_target, _pitch_target);
+}
+
+// get_lean_angles_to_accel - convert roll, pitch lean angles to lat/lon frame accelerations in cm/s/s
+void AC_PosControl::accel_to_lean_angles(float accel_x_cmss, float accel_y_cmss, float& roll_target, float& pitch_target) const
+{
+    float accel_right, accel_forward;
 
     // rotate accelerations into body forward-right frame
-    accel_forward = accel_target_filtered.x*_ahrs.cos_yaw() + accel_target_filtered.y*_ahrs.sin_yaw();
-    accel_right = -accel_target_filtered.x*_ahrs.sin_yaw() + accel_target_filtered.y*_ahrs.cos_yaw();
+    accel_forward = accel_x_cmss*_ahrs.cos_yaw() + accel_y_cmss*_ahrs.sin_yaw();
+    accel_right = -accel_x_cmss*_ahrs.sin_yaw() + accel_y_cmss*_ahrs.cos_yaw();
 
     // update angle targets that will be passed to stabilize controller
-    _pitch_target = constrain_float(atanf(-accel_forward/(GRAVITY_MSS * 100))*(18000/M_PI),-lean_angle_max, lean_angle_max);
-    float cos_pitch_target = cosf(_pitch_target*M_PI/18000);
-    _roll_target = constrain_float(atanf(accel_right*cos_pitch_target/(GRAVITY_MSS * 100))*(18000/M_PI), -lean_angle_max, lean_angle_max);
+    pitch_target = atanf(-accel_forward/(GRAVITY_MSS * 100))*(18000/M_PI);
```

Comment:
Shouldn't this be 100.0f and 18000.0f ?

Reply:
I think the "100" and "18000" are converted to floats as part of the divid but I've added the .0f at the end in any case, txs.